pkg/resources/object_renaming_lists_and_sets_acceptance_test.go

```diff
@@ -0,0 +1,463 @@
+					{"string": "333", "int": 333},
+					{"string": "333", "int": 333},
+				}),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					assert.HasListItemsOrderIndependent("snowflake_object_renaming.test", "list", []map[string]string{
+						{"string": "111", "int": "111"},
+						{"string": "111", "int": "111"},
+						{"string": "222", "int": "222"},
+						{"string": "222", "int": "222"},
+						{"string": "333", "int": "333"},
+						{"string": "333", "int": "333"},
+					}),
+				),
+			},
+		},
+	})
+}
+
+func objectRenamingConfigList(listItems []map[string]any) string {
+	generateListItem := func(name *string, s string, i int) string {
+		var nameField string
+		if name != nil {
+			nameField = fmt.Sprintf(`name = "%s"`, *name)
+		}
+		return fmt.Sprintf(`
+list {
+	%[1]s
+	string = "%[2]s"
+	int = %[3]d
+}
+`, nameField, s, i)
+	}
+
+	generatedListItems := ""
+	for _, item := range listItems {
+		var name *string
+		if nameValue, ok := item["name"]; ok && nameValue != nil {
+			name = sdk.String(nameValue.(string))
+		}
+		generatedListItems += generateListItem(name, item["string"].(string), item["int"].(int))
+	}
+
+	return fmt.Sprintf(`
+resource "snowflake_object_renaming" "test" {
+	%s
+}
+`, generatedListItems)
+}
+
+func TestAcc_BasicManuallyOrderedListFlow(t *testing.T) {
+	_ = testenvs.GetOrSkipTest(t, testenvs.EnableAcceptance)
+	_ = testenvs.GetOrSkipTest(t, testenvs.EnableObjectRenamingTest)
+	acc.TestAccPreCheck(t)
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: objectRenamingConfigManuallyOrderedList([]map[string]any{
+					{"name": "name1", "order": 1},
+					{"name": "name2", "order": 2},
+					{"name": "name3", "order": 3},
+				}),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_object_renaming.test", "manually_ordered_list.0.name", "name1"),
+					resource.TestCheckResourceAttr("snowflake_object_renaming.test", "manually_ordered_list.1.name", "name2"),
+					resource.TestCheckResourceAttr("snowflake_object_renaming.test", "manually_ordered_list.2.name", "name3"),
+				),
+			},
+			// Change values
+			{
+				Config: objectRenamingConfigManuallyOrderedList([]map[string]any{
+					{"name": "name2", "order": 1},
+					{"name": "name2", "order": 2},
+					{"name": "name4", "order": 3},
+				}),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_object_renaming.test", "manually_ordered_list.0.name", "name2"),
+					resource.TestCheckResourceAttr("snowflake_object_renaming.test", "manually_ordered_list.1.name", "name2"),
+					resource.TestCheckResourceAttr("snowflake_object_renaming.test", "manually_ordered_list.2.name", "name4"),
+				),
+			},
+			// Change the order
```

Comment:
Change the order in config but not change the desired order of items

Reply:
👍 changed in the new test